src/core/qgsuserprofile.cpp

```diff
@@ -0,0 +1,114 @@
+#include "qgsuserprofile.h"
+#include "qgsapplication.h"
+
+#include <QDir>
+#include <QTextStream>
+#include <QSettings>
+#include <QSqlDatabase>
+#include <QSqlQuery>
+
+QgsUserProfile::QgsUserProfile( QString folder )
+{
+  mProfileFolder = folder;
+}
+
+const QString QgsUserProfile::folder() const
+{
+  return mProfileFolder;
+}
+
+QgsError QgsUserProfile::validate() const
+{
+  QgsError error;
+  if ( !QDir( mProfileFolder ).exists() )
+  {
+    error.append( QObject::tr( "Profile folder doesn't exist" ) );
```

Comment:
might be a good idea to include the missing folder name here too (helps with debugging)